syntax="proto3";
import "server.proto";

service Register{
    rpc registerReplica (registerReplicaRequest) returns (registerReplicaResponse) {}
    rpc GetReplicaList (ClientRequest) returns (ClientResponse) {}
}

message registerReplicaRequest{
    Address address=1;
}

message registerReplicaResponse{
    string status=1;
    Address paddress=2;
}

message ClientRequest{
    string uuid=1;
}

message ClientResponse{
    string status=1;
    repeated Address addresses=2;
}

message RegistryServer{
    repeated Address replicas=1;
    Address p_address=2;

}


